### PR TITLE
[4.0 dev] Correct the layout of the modals in "Manage Folders" and "New File" in Template manager.

### PR DESCRIPTION
**Pull request by particthistle**
Pull Request for Issue #31033 and #30954.

### Summary of Changes
Modified the modal settings for both the "Manage Folders" And "New File" modal windows that appear when you edit a template.
- Modal size changed from 50vh and 60vw to 70vh and 80vw
- Layout of modal columns changed from col-md-6 & col-md-6 to col-md-4 and col-md-8 to closer match Templates: Customisation screen ratios (it uses 3/9)
- Class for .modal-body .tree-holder - sets a max-height for the tree-holder in the modal view. The .tree-holder class is also applied in the customisation screen view, and doesn't need to have it's height restricted there.
- Several responsive styles have been added to reformat the modal contents on max-width:768px

### Testing Instructions
- Familiarise yourself with the BEFORE status.
- Apply the patch. _Note that you can manually add the new class to \media\com_templates\css\admin-templates-default.css rather than needing to install the patch using NPM as the CSS file is precompiled._
- Go to: System > Site Templates > Edit the template by clicking on "Cassiopeia Details and Files"
- Click on the "New File" and "Manage Folders" buttons.
- Check what the layout of the modal looks like.
- Go to: System > Adminstrator Templates > Edit the template by clicking on "Atum Details and Files"
- Click on the "New File" and "Manage Folders" buttons.
- Check what the layout of the modal looks like.

### Actual result BEFORE applying this Pull Request
Per the various screen grabs in #31033, you before applying the PR you will have issues on screens under 1400px with the layout being cramped, and on all screens the folder tree when displayed in the modal, the tree extends over the bottom of the modal, meaning the close and/or delete buttons (depending on the screen) are disabled due to the z-index of the folder tree.
image

### Expected result AFTER applying this Pull Request
_Note that the CSS change is required to make the folder tree height correct._
- Layout will change so the columns are 4:8 ratio.
- With the folder tree height capped proportional to the new height of the modal, the buttons now work.
image

- Check also responsive layout on max-width:768px and below. 
- Uncommon as it would be for someone to modify the template via mobile, the layout now at least provides the ability to do so if needed.
31037 Responsive
### Documentation Changes Required
No documentation changes are needed.

### CODE REVIEW

**Comment by particthistle** [comment]
Further changes are required in admin-templates-default.css to add responsive css to the modal as there's additional layout problems with the modal on smaller screens. That CSS file currently has no breakpoints. 

Inspecting the code, I need to add a breakpoint for: breakpoint-md: 768px;

image

**Comment by particthistle** [comment]
CSS for mobile partially fixed. New styling fixes the Manage Folders view (see screenshot) but not the new file view, as that requires two options: either create the new file, or upload a file (though hard to do from a phone). Look forward to hearing thoughts.

image

**Comment by infograf768** [comment]
Almost there.
Problems with `max-height` in
```
@media (min-width: 768px) {
	.modal-body .tree-holder {
		max-height: 30vh;  // Here, I had to comment it.
		margin-bottom: 20px;
	}
}
```

template_files

Macintosh Firefox, Safari

**Comment by particthistle** [comment]
```
@media (max-width: 768px) {
	.modal-title { /* Reduces the modal title to stop it wrapping */
		font-size: 1rem;
	}
	.modal-body .tree-holder { /* Shrinks area for the tree in mobile modal view */
		max-height: 12vh;
		margin-bottom: 20px;
	}
	.modal-body .card-body { /* Padding in the card boxes where the filename or folder name is created */
		padding: .75rem;
	}
	.modal-body .col-md-4, .modal-body .col-md-8 { /* Fills the modal wider when the columns stack. */
		padding-left: 0;
		padding-right: 0;
	}
	.card-body .mt-2 { /* Reduces the size of the "Maximum Upload size note" */
		font-size: 12px;
	}
}
```
Responsive styling improved. I've put in changes specific to these modals, rather than applying it to overall responsive classes in the template, though some of that might need to be considered globally in the template main template.css file.

_CSS Comments above are not included in the admin-templates-default.css._
31037 Responsive

File: build/media_source/com_templates/css/admin-templates-default.css [hunk 1]
```diff
@@ -27,6 +27,29 @@
...
+		font-size: 1rem;
+	}
+	.modal-body .tree-holder {
+		max-height: 12vh;
+		margin-bottom: 20px;
+	}
+	.modal-body .card-body {
+		padding: .75rem;
+	}
+	.modal-body .col-md-4, .modal-body .col-md-8 {
```
Comment (Quy) [inline build/media_source/com_templates/css/admin-templates-default.css]:
```suggestion
	.modal-body .col-md-4,
	.modal-body .col-md-8 {
```

**Comment by nadjak77** [comment]
I have tested this item :red_circle: unsuccessfully on 5cfbf5e2e22dc639b33e7a9d800661689b351f32<br /><br />I still have the problem with the overlapping formfields (see comments)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31037">issues.joomla.org/tracker/joomla-cms/31037</a>.</sub>

**Comment by particthistle** [comment]
@nadjak77 your comments didn't arrive. Can you update what is happening for you?

I checked it again a moment ago, and was all working fine (Firefox on Windows) so some more details on what you're encountering are needed.

**Comment by nadjak77** [comment]
I have tested this item :white_check_mark: successfully on 5cfbf5e2e22dc639b33e7a9d800661689b351f32<br /><br />tested again and now it works. I had some trouble with the cache (deleting the browser cache did not solve the problem).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31037">issues.joomla.org/tracker/joomla-cms/31037</a>.</sub>

**Comment by webfeuerflo** [comment]
I have tested this item ✅ successfully if I replace the min-file manually. So if the min-file is rebuilt, everything should be fine!

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31037">issues.joomla.org/tracker/joomla-cms/31037</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**